ios/objective-c/v2/AmpliObjectiveCSampleApp/AmpliObjectiveCSampleApp/AppDelegate.m

```diff
@@ -0,0 +1,167 @@
+
+@end
+
+@implementation AppDelegate
+
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    /*
+     * '[Ampli instance]' is the default instance of Ampli
+     *
+     * When you pull your tracking plan you can use the defaults and call load() without arguments
+     * This requires connecting your account via `ampli pull` which will set you API key in the generated Ampli SDK
+     */
+    // [Ampli.instance]
+
+    /*
+     * OR Specify a AmpliEnvironment
+     */
+    // [Ampli.instance load:[LoadOptions initWithEnvironment:dev]];
+
+    /*
+     * OR Provide a specific Amplitude API key
+     */
+    // [Ampli.instance load:[LoadOptions initWithApiKey:@"Custom api key"]];
+
+    /*
+     * OR Use an existing Amplitude instance
+     * requires `@import AmplitudeSwift;`
+     */
+    // AMPConfiguration* configuration = [AMPConfiguration initWithApiKey:@"Custom api key" instanceName:@"instanceName"];
+    // Amplitude* client = [Amplitude initWithConfiguration:configuration];
+    // LoadClientOptions* clientOptions = [LoadClientOptions initWithInstance:client];
+    // [Ampli.instance load:[LoadOptions initWithClientOptions:clientOptions]];
+
+    /*
+     * For testing you can disable Ampli
+     */
+    // [Ampli.instance load:[LoadOptions initWithDisabled:ENV.IS_TESTING ? YES : NO]];
+
+    /*
+     * Make as many Ampli instances as you want
+     */
+    // Ampli* ampli2 = [Ampli new];
+    // [ampli2 load:[LoadOptions initWithApiKey:@"Custom api key"]];
+
+    NSString *apiKey = [[[NSProcessInfo processInfo] environment] objectForKey:@"AMPLITUDE_API_KEY"];
+    Ampli *ampli = [Ampli instance];
+
+    // Load
+    [ampli load:[LoadOptions initWithApiKey:apiKey]];
+
+    // Add Plugin
+    [ampli.client add:[AMPPlugin initWithType:AMPPluginTypeDestination execute:^AMPBaseEvent* _Nullable(AMPBaseEvent* _Nonnull event) {
+        NSString *logString = [NSString stringWithFormat:@"event_type=%@", event.eventType];
+        NSLog(@"[ampli] %@", logString);
+        return event;
+    }]];
+
+    // Identify
+    [ampli identify:@"ampli-objc-user" identify:[Identify requiredNumber: 1.23F builderBlock:^(IdentifyBuilder *b) {
+        b.optionalArray = [NSArray arrayWithObjects:@"optional string", nil];
+    }]];
+
+    // Set Group
+    [ampli.client setGroup:@"ampli group type" groupName:@"ampli objective-c group"];
+
+    // GroupIdentify
+    AMPIdentify *identifyArgs = [AMPIdentify new];
+    [identifyArgs set:@"requiredBoolean" value:@true];
+    [identifyArgs set:@"optionalString" value:@"optional string"];
+    [ampli.client groupIdentify:@"ampli group type" groupName:@"ampli objective-c group" identify:identifyArgs];
+
+    // Track events with dedicated event methods
+    [ampli track:[EventNoProperties build]];
+
+    [ampli track:[EventMaxIntForTest intMax10: 5]];
+    [ampli track:[EventMaxIntForTest intMax10: 20]];
+
+    [ampli track:[EventWithConstTypes build]];
```

Comment:
nit: can we rename `build` `new`? I feel that might be a little more idiomatic. wdyt?

Reply:
we can but XCode/compiler shows a warning - `new` is expected to return its class type
![image](https://github.com/amplitude/ampli-examples/assets/9509929/c7e16508-56a2-461b-a039-120685142e4a)

Reply:
Thanks for checking. `build` is good.